rpc/block_svc_routines.c

```diff
@@ -41,7 +41,8 @@
                                 "'%s' ls | grep -e tpg -e '%s' | grep -B1 '%s' | grep -o 'tpg\\w'"
 # define   GB_CHECK_PORTAL      "targetcli /iscsi/" GB_TGCLI_IQN_PREFIX \
                                 "'%s' ls | grep '%s' > " DEVNULLPATH
-# define   GB_SAVECONFIG_SO_CHECK "grep -m 1 '\"config\":.*%s\",' " GB_SAVECONFIG " > " DEVNULLPATH
+# define   GB_SAVECFG_GBID_CHECK "grep -m 1 '\"config\": .*%s\",' " GB_SAVECONFIG " > " DEVNULLPATH
+# define   GB_SAVECFG_NAME_CHECK "grep -cs '\"name\": \"%s\",' " GB_SAVECONFIG " " GB_SAVECONFIG_TEMP
```

Comment:
@lxbsz why -sm 1 is not enough ?

Reply:
if we use -sm 1 with grep then we can directly use gbRunner() and get the return value, so we don't have to deal with strings, refer blockCheckBlockLoadedStatus()

Reply:
```
[2019-07-12 10:52:34.151381] INFO: command [grep -sm 1 '"name": "block12",' /etc/target/saveconfig.json /etc/target/saveconfig.json.temp], childExitStatus [512] [at utils.c+654 :<gbRunner>]
[2019-07-12 10:52:34.151484] INFO: gbRunner(exec) [2] [at block_svc_routines.c+4393 :<block_create_common>]
```

It will always return 2.

Did I miss something important ?

Thanks.